### DISCUSSION TITLE
feature: make edgy resize step configurable

### DISCUSSION THREAD

**Discussion by marcinjahn** [LazyVim/LazyVim#3777] [Extra Requests]
The default resize step of edgy is set to "2". It's quite small, and requires lots of key presses to actually get to any usable size in some cases.
It seems that changing this configuration would, at the moment, require overwriting of the whole `opts` function, which has much more stuff than just resize size configuration.

I think it'd be really nice if the resize step could be configured in some easier way.

**Comment by iamxiaojianzheng** [comment]
try this.

```lua
return {
  {
    "folke/edgy.nvim",
    opts = function(_, opts)
      local step = 15
      opts.keys = {
        -- increase width
        ["<c-Right>"] = function(win)
          win:resize("width", step)
        end,
        -- decrease width
        ["<c-Left>"] = function(win)
          win:resize("width", 0 - step)
        end,
        -- increase height
        ["<c-Up>"] = function(win)
          win:resize("height", 2)
        end,
        -- decrease height
        ["<c-Down>"] = function(win)
          win:resize("height", -2)
        end,
      },
    end
  }
}
```

**Reply by marcinjahn** [reply]
Thanks, that works. I wonder, does the "original" config from LazyVim (https://www.lazyvim.org/extras/ui/edgy) still get executed and then, your patch is executed on top of it?

**Reply by dpetka2001** [reply]
This just adds on top of the default LazyVim config.
